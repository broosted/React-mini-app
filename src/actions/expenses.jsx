import uuid from 'uuid';

//Add_ Expens
export const addExpense = ({
    description = '',
    note = '',
    amount = 0,
    createdAt = 0 } = {}
) => ({
    type: 'ADD_EXPENSE',
    expense : {
        id: uuid(),
        description,
        note,
        amount,
        createdAt
    }

});
//REmove_expnes

export const removeExpense = ({id } = {}) => ({
    type: 'REMOVE_EXPENSE',
        id
});

//edit_expense

export const editExpense = (id , updates) => ({
    type: 'EDIT_EXPENSE',
    id,
    updates

});

