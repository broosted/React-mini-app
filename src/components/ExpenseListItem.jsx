import React from 'react';
import {connect} from 'react-redux';

import {removeExpense} from '../actions/expenses.jsx';
import selectExpenses from '../selectors/expenses.jsx';
import { BrowserRouter, Route, Switch, Link, NavLink} from 'react-router-dom';


const ExpenseListItem = ({id, description, note, amount, createdAt, props}) =>  (
   <li>Description:
       <Link to={`/edit/${id}`} >{description}</Link> 
       <p>Note: {note}</p>
       <p>Amount: {amount}</p>
       <p>CreatedAt: {createdAt}</p>
       <button onClick={() => {
           props.dispatch(removeExpense({id}));
       }}>Remove</button>
       
   </li>
);




export default connect()(ExpenseListItem);
