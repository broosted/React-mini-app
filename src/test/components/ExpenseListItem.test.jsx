import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';

import ExpenseListItem from '../../components/ExpenseListItem.jsx';
import expenses from '../fixed/expenses.js';
import toJSON from 'enzyme-to-json';



const mockStore = configureMockStore([])


test('should render ExpenseList with expenses', () => {
    const store = mockStore({})

    const wrapper = shallow(
        <Provider store={store}>
    <ExpenseListItem expenses={expenses[1]} />
    </Provider>
    )

    expect(wrapper).toMatchSnapshot()
})


