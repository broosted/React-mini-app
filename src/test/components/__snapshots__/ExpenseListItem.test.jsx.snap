// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseList with expenses 1`] = `
<Connect(ExpenseListItem)
  expenses={
    Object {
      "amount": 109500,
      "createdAt": -345600000,
      "description": "Rent",
      "id": "2",
      "note": "",
    }
  }
/>
`;
